bundles/org.smarthomej.binding.tuya/src/main/java/org/smarthomej/binding/tuya/internal/handler/TuyaDeviceHandler.java

```diff
@@ -300,12 +306,47 @@ public void handleCommand(ChannelUID channelUID, Command command) {
             if (command instanceof OnOffType) {
                 commandRequest.put(configuration.dp, OnOffType.ON.equals(command));
             }
+        } else if (CHANNEL_TYPE_UID_IR_CODE.equals(channelTypeUID)) {
+            if (command instanceof StringType) {
+                if (configuration.irType.equals("base64")) {
+                    commandRequest.put(1, "study_key");
+                    commandRequest.put(7, command.toString());
+                } else if (configuration.irType.equals("tuya-head")) {
+                    if (configuration.irCode != null && !configuration.irCode.isEmpty()) {
+                        commandRequest.put(1, "send_ir");
+                        commandRequest.put(3, configuration.irCode);
+                        commandRequest.put(4, command.toString());
+                        commandRequest.put(10, configuration.irSendDelay);
+                        commandRequest.put(13, configuration.irCodeType);
+                    } else {
+                        logger.error("irCode is not set");
```

Comment:
```suggestion
                        logger.warn("irCode is not set for channel {}", channelUID);
```

ERROR is reserved for conditions where the stability of the whole system is in danger.